### PR TITLE
[THREESCALE-10130] APIcast using stale configuration for a deleted Product

### PR DESCRIPTION
**Pull request by tkan145**
## What
Fix https://issues.redhat.com/browse/THREESCALE-10130

## Verification steps
* Install 3scale and create two Product A and B
* Checkout this branch
* Start development environment
```
make development
make dependencies
```
* Start APIcast with `THREESCALE_PORTAL_ENDPOINT=https://<token>@<provider domain>`
```
THREESCALE_DEPLOYMENT_ENV=staging APICAST_LOG_LEVEL=debug APICAST_WORKER=1 APICAST_CONFIGURATION_LOADER=boot APICAST_CONFIGURATION_CACHE=60 THREESCALE_PORTAL_ENDPOINT=https://token@3scale-admin.example.com ./bin/apicast
```
* Send requests
```
# capture apicast IP
APICAST_IP=$(docker inspect apicast_build_0-development-1 | yq e -P '.[0].NetworkSettings.Networks.apicast_build_0_default.IPAddress' -)

curl -v -k -H "Host: example.productA.com:443" -H "Accept: application/json" http://${APICAST_IP}:8080/user_key=<user_key>

curl -v -k -H "Host: example.productB.com:443" -H "Accept: application/json" http://${APICAST_IP}:8080/user_key=<user_key>
```
* Both responses should be `HTTP/1.1 200 OK`
* Now visit 3scale admin portal and delete ProductB.
* Wait for APIcast to reload the configuration
* Send request to ProductA, response should be `HTTP/1.1 200 OK`
* Send request to ProductB, response should be `HTTP/1.1 404 Not Found`
```
< HTTP/1.1 404 Not Found
< Server: openresty
< Date: Tue, 23 Jul 2024 04:41:21 GMT
< Content-Type: text/plain
< Transfer-Encoding: chunked
< Connection: keep-alive
```

### CODE REVIEW

**Comment by tkan145** [comment]
@kevprice83 can you also have a look at this PR?

Please ignore the CHANGELOG conflict, I will rebase later

File: gateway/src/apicast/configuration_loader.lua [hunk 1]
```diff
@@ -229,7 +235,7 @@ local function lazy_load_config(configuration, host)
     if not config then
       ngx.log(ngx.WARN, 'failed to get config for host: ', host)
     end
-    _M.configure(configuration, config)
+    _M.configure(configuration, config, host)
```
Comment (eguzki) [inline gateway/src/apicast/configuration_loader.lua:238]: `host` ? shouldn't be `true`?

**Comment by eguzki** [comment]
verification steps working as expected

File: gateway/src/apicast/configuration_loader.lua (hunk 1 above)
Comment (tkan145) [reply gateway/src/apicast/configuration_loader.lua]: Removed. Lazy mode will not return the stale configuration so we don't really need to clear the cache.

File: gateway/src/apicast/configuration_store.lua [hunk 2]
```diff
@@ -165,16 +165,18 @@ function _M.store(self, config, ttl)
   return config
 end
 
-function _M.reset(self, cache_size)
```
Comment (eguzki) [inline gateway/src/apicast/configuration_store.lua:168]: I assume the cache_size is never being used

**Review by eguzki — APPROVED** [review]

**Pull request merged**